RawSpeed/TiffIFD.cpp

```diff
@@ -61,6 +61,11 @@ TiffIFD::TiffIFD(FileMap* f, uint32 offset, uint32 _depth) {
       t = new TiffEntry(f, entry_offset, offset);
     } catch (IOException) { // Ignore unparsable entry
       continue;
+    } catch (...) {
+      // make sure all already allocated mEntries are freed again
+      // will be gone after cleanup using unique_ptr containers
+      this->~TiffIFD();
```

Comment:
Please ignore my link to b453d047dfcbb74378b9caabe0258fffd315da4a. I guess I should take a break and eat something before blaming people of stuff they clearly did not do ;).